runbotics/runbotics-desktop/src/action/sharepoint/excel/sharepoint-excel.action-handler.ts

```diff
@@ -1,76 +1,64 @@
 import { Injectable } from '@nestjs/common';
-import { StatelessActionHandler, DesktopRunResponse } from 'runbotics-sdk';
+import { StatelessActionHandler } from 'runbotics-sdk';
 
-import { MicrosoftService, CloudPath, MicrosoftSession } from '#action/microsoft';
 import * as SharepointTypes from './types';
+import { ExcelService } from '../../microsoft/excel/excel.service';
 
 @Injectable()
 export default class SharepointExcelActionHandler extends StatelessActionHandler {
     constructor(
-        private readonly microsoftSession: MicrosoftSession,
-        private readonly microsoftService: MicrosoftService,
+        private readonly excelService: ExcelService
     ) {
         super();
     }
 
-    async openFileFromSite(
-        input: SharepointTypes.SharepointOpenFromSiteActionInput
+    async openFile(
+        input: SharepointTypes.SharepointExcelOpenFileActionInput
     ): Promise<SharepointTypes.SharepointOpenActionOutput> {
-        const cloudPath = CloudPath.SITE;
-        const token = await this.microsoftSession.getToken();
-        const sharepointSiteId = await this.microsoftService.getSiteIdByPath(token.token, input.siteRelativePath);
-        const sharepointDriveId = await this.microsoftService.getDriveId(token.token, sharepointSiteId, input.listName);
-        const sharepointFileId = await this.microsoftService.getFileIdByPath(token.token, cloudPath, input.filePath);
-        const sharepointWorksheetId = await this.microsoftService.getWorksheetId(token.token, cloudPath, input.worksheetName);
-        return await this.microsoftService.createSession(token.token, cloudPath, input.persistChanges);
-    }
-
-    async openFileFromRoot(
-        input: SharepointTypes.SharepointOpenFromRootActionInput
-    ): Promise<SharepointTypes.SharepointOpenActionOutput> {
-        const cloudPath = CloudPath.ROOT;
-        const token = await this.microsoftSession.getToken();
-        const sharepointFileId = await this.microsoftService.getFileIdByPath(token.token, cloudPath, input.filePath);
-        const sharepointWorksheetId = await this.microsoftService.getWorksheetId(token.token, cloudPath, input.worksheetName);
-        return await this.microsoftService.createSession(token.token, cloudPath, input.persistChanges);
+        return await this.excelService.openFile(
+            //eslint-disable-next-line
+            //@ts-ignore
+            input.platform,
```

Comment:
no need for ts check ignore if you pass type `'OneDrive' | 'SharePoint'` instead of a more broad string type